napari/_qt/layer_controls/qt_image_controls.py

```diff
@@ -98,8 +98,12 @@ def __init__(self, layer):
         self.grid_layout.addWidget(self.contrastLimitsSlider, 1, 1)
         self.grid_layout.addWidget(QLabel('gamma:'), 2, 0)
         self.grid_layout.addWidget(self.gammaSlider, 2, 1)
-        self.grid_layout.addWidget(QLabel('colormap:'), 3, 0)
-        self.grid_layout.addLayout(colormap_layout, 3, 1)
+
+        # The colormap widget won't be created when image has a rgb layer
+        if layer.rgb is False:
+            self.grid_layout.addWidget(QLabel('colormap:'), 3, 0)
```

Comment:
let's actually store the reference to the colormap text label explicitly by making `self.colormapLabel = QLabel('colormap:')`. Can you do that here https://github.com/napari/napari/blob/master/napari/_qt/layer_controls/qt_image_controls_base.py#L56

And then here can call `self.grid_layout.addWidget(self.colormapLabel, 3, 0)`

You can also then change this line accordingly too https://github.com/napari/napari/blob/master/napari/_qt/layer_controls/qt_surface_controls.py#L39

Having this reference to the QLabel will make hiding the label much cleaner